src/lib.rs

```diff
@@ -467,6 +473,44 @@ fn apply_patches(target: &str, inner: &Path) {
         .unwrap();
 }
 
+fn apply_patches_aarch64_apple_darwin(target: &str, inner: &Path) {
+    if target != "aarch64-apple-darwin" {
+        return;
+    }
+
+    // Apply build system changes to allow configuring and building
+    // for Apple's ARM64 platform.
+    // https://github.com/openssl/openssl/pull/12369
+
+    let mut buf = String::new();
+    let path = inner.join("Configurations/10-main.conf");
+    File::open(&path).unwrap().read_to_string(&mut buf).unwrap();
+
+    assert!(
+        !buf.contains("darwin64-arm64-cc"),
+        "{} already contains instructions for aarch64-apple-darwin",
+        path.display(),
+    );
+
+    const PATCH: &'static str = r#"
+    "darwin64-arm64-cc" => {
+        inherit_from     => [ "darwin-common", asm("aarch64_asm") ],
+        CFLAGS           => add("-Wall"),
+        cflags           => add("-arch arm64"),
+        lib_cppflags     => add("-DL_ENDIAN"),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG",
+        perlasm_scheme   => "ios64",
+    },"#;
+
+    let x86_64_stanza = buf.find(r#"    "darwin64-x86_64-cc""#).unwrap();
+    buf.insert_str(x86_64_stanza, PATCH);
+
+    File::create(&path)
```

Comment:
This can just use `fs::write(&path, buf.as_bytes())`

Reply:
Ok, I just copied the existing patch structure and assumed that the library was striving to support older Rust versions. 

Would you like me to clean up the MUSL to use these functions as well?

Reply:
The oldest version we need to support is 1.31 (at least that's what rust-openssl tests against), so the fs:: methods should be fine there. Yeah, might as well clean up the MUSL one as well.